src/main/ApplicationImpl.cpp

```diff
@@ -549,27 +549,109 @@ ApplicationImpl::joinAllThreads()
     LOG(DEBUG) << "Joined all " << mWorkerThreads.size() << " threads";
 }
 
-bool
-ApplicationImpl::manualClose()
+std::string
+ApplicationImpl::manualClose(optional<uint32_t> const& manualLedgerSeq,
+                             optional<TimePoint> const& manualCloseTime)
 {
     // Manual close only makes sense for validating nodes
     if (!mConfig.NODE_IS_VALIDATOR)
     {
-        return false;
+        throw std::invalid_argument(
+            "Issuing a manual ledger close requires NODE_IS_VALIDATOR to be "
+            "set to true.");
     }
 
     if (mConfig.MANUAL_CLOSE)
     {
-        mHerder->triggerNextLedger(mLedgerManager->getLastClosedLedgerNum() + 1,
-                                   true);
-        return true;
+        auto const startLedgerSeq = getLedgerManager().getLastClosedLedgerNum();
+        if (startLedgerSeq == std::numeric_limits<uint32_t>::max())
+        {
+            throw std::invalid_argument(
+                fmt::format(FMT_STRING("Manually closed ledger "
+                                       "sequence number already at max ({})"),
+                            startLedgerSeq));
+        }
+        auto minLedgerSeq = startLedgerSeq + 1;
+        auto const targetLedgerSeq =
+            manualLedgerSeq ? *manualLedgerSeq : minLedgerSeq;
+        if (targetLedgerSeq < minLedgerSeq)
+        {
+            throw std::invalid_argument(
+                fmt::format(FMT_STRING("Invalid manual close ledger "
+                                       "sequence number {} (minimum {})"),
+                            targetLedgerSeq, minLedgerSeq));
+        }
+        auto const lastCloseTime = getLedgerManager()
+                                       .getLastClosedLedgerHeader()
+                                       .header.scpValue.closeTime;
+        auto const now = VirtualClock::to_time_t(getClock().system_now());
+        uint32_t const defaultManualCloseTimeInterval =
+            getConfig().getExpectedLedgerCloseTime().count();
+        auto nextCloseTime = lastCloseTime + 1;
```

Comment:
It turns out it's messier than either:  `HerderImpl::triggerNextLedger()` uses `VirtualClock::to_time_t()` on the `closeTime`, so we need to do the same to validate against the last closed header's `closeTime`, and possibly to produce a `closeTime` derived from that.  Then we convert it back to a `system_time_point` to call `setCurrentVirtualTime()`.  So a `closeTime` that overflows is one that isn't stable under all that converting.  That's nine-point-something billion.  I think that the best way to check for that is to do that stability test itself, so that's what the code does now.